packages/assets-controllers/src/TokenDetectionController.test.ts

```diff
@@ -141,12 +175,25 @@ describe('TokenDetectionController', () => {
         providerConfig,
       });
     });
+
+    controllerMessenger.unregisterActionHandler(
```

Comment:
Instead of having to unregister the action handler, what do you think about extracting the mock function used as the action handler to a variable, using it on line 251, and then calling `.mockReturnValue` to override its return value here?

Reply:
Nice! Applied here: https://github.com/MetaMask/core/pull/3609/commits/7d1116098d147af862a078812e2e6c0a3bd465de